compileoptions.cmake

```diff
@@ -36,7 +36,8 @@ if (CLR_CMAKE_PLATFORM_UNIX)
   # There are constants of type BOOL used in a condition. But BOOL is defined as int
   # and so the compiler thinks that there is a mistake.
   add_compile_options(-Wno-constant-logical-operand)
-
+  add_compile_options(-Wno-null-dereference)
```

Comment:
I am not currently using Clang 3.9.  This was from November.  I saw some patches fixing warnings, but I never got around to fixing them.

It may take a while for me to figure out whether these are still required.

I would suggest abandoning this patch